src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -158,7 +158,18 @@ public class JsonInputFormatter : TextInputFormatter
 
                     if (successful)
                     {
-                        return InputFormatterResult.SuccessAsync(model);
+                        if (model == null && !context.AllowEmptyInput)
```

Comment:
> The original issue (#4750) was due to someone getting tripped up by receiving `null` when `ModelState.IsValid` was `true`,

The original case is already handled by your changes in `InputFormatter`. The client posted no content, not the JSON `null` keyword. As I said, let's discuss going beyond the `ContentLength == 0` check in a new issue. Checking for whitespace or treating everything the JSON deserializer converts to `null` as if the client posted nothing is separable and controversial, at least to me.